src/vas.jl

```diff
@@ -49,21 +49,28 @@ This puts together the model and a sampler. This is what we think of as
 a simulation. We will organize this like it's a finite state machine,
 so it will have an initializer, a dynamics, and an observer.
 """
-mutable struct VectorAdditionFSM
+mutable struct VectorAdditionFSM{
+        VasT <: VectorAdditionSystem, 
+        VasS <: VectorAdditionState,
+        S,
+        R <: AbstractRNG,
+        I <: Function,
+        O <: Function
+    }  
```

Comment:
Making a template parameter for everything, huh? I figured there was a diminishing return, but I haven't looked at timings. I trust you on this.

Reply:
It surprised me too to see compiler inference fail, but the proof is in the pudding.